core/templates/dev/head/components/SolutionEditorDirective.js

```diff
@@ -28,15 +28,27 @@ oppia.directive('solutionEditor', [
         getOnSaveFn: '&onSave'
       },
       templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
-        '/components/' + 'solution_editor_directive.html'),
+        '/components/solution_editor_directive.html'),
       controller: [
-        '$scope', 'editabilityService', function($scope, editabilityService) {
+        '$scope', 'editabilityService', 'stateSolutionService',
+        function($scope, editabilityService, stateSolutionService) {
           $scope.isEditable = editabilityService.isEditable();
 
           $scope.editSolutionForm = {};
           $scope.solutionEditorIsOpen = false;
+          $scope.solutionCorrectAnswerHtml = '';
 
-          $scope.solutionMemento = null;
+          $scope.getCorrectAnswerHtml = function() {
+            $scope.solutionCorrectAnswerHtml = (
```

Comment:
I'm not sure you need a separate var for this. Why not reference solution.getCorrectAnswerHtml() directly when you need it? You could store the interaction and object type in the Solution object if that makes things easier.